tree/dataframe/src/RDFInterfaceUtils.cxx

```diff
@@ -51,6 +54,40 @@ namespace RDF {
 class RActionBase;
 
 
+HeadNode_t CreateSnaphotRDF(const ColumnNames_t &validCols,
+                            const std::string &fullTreeName,
+                            const std::string &fileName,
+                            bool isLazy,
+                            RLoopManager &loopManager,
+                            std::unique_ptr<RDFInternal::RActionBase> actionPtr)
+{
+   // create new RDF
+   ::TDirectory::TContext ctxt;
+   // Now we mimic a constructor for the RDataFrame. We cannot invoke it here
+   // since this would introduce a cyclic headers dependency.
+
+   // Keep these two statements separated to work-around an ABI incompatibility
+   // between clang (and thus cling) and gcc in the way std::forward is handled.
+   // See https://sft.its.cern.ch/jira/browse/ROOT-9236 for more detail.
+   auto rlm_ptr = std::make_shared<RLoopManager>(nullptr, validCols);
+   auto snapshotRDF = std::make_shared<RInterface<RLoopManager>>(rlm_ptr);
+   auto chain = std::make_shared<TChain>(fullTreeName.c_str());
+   chain->Add(std::string(fileName).c_str());
+   snapshotRDF->fProxiedPtr->SetTree(chain);
+   auto snapshotRDFResPtr = MakeResultPtr(snapshotRDF, loopManager, std::move(actionPtr));
```

Comment:
At this point I don't think this dance is required anymore: we should just construct a `RDataFrame(chain)` here

Reply:
Of course.